zerver/actions/message_edit.py

```diff
@@ -598,20 +576,15 @@ def user_info(um: UserMessage) -> dict[str, Any]:
             "flags": um.flags_list(),
         }
 
-    if target_message.is_stream_message():
-        if topic_name is not None:
-            new_topic_name = topic_name
-        else:
-            new_topic_name = target_message.topic_name()
-
+    if isinstance(message_edit_request, StreamMessageEditRequest):
         stream_topic: StreamTopicTarget | None = StreamTopicTarget(
-            stream_id=stream_id,
-            topic_name=new_topic_name,
+            stream_id=message_edit_request.orig_stream.id,
+            topic_name=message_edit_request.topic_name,
```

Comment:
Why is this a combination of the original stream and new topic?

Reply:
This is only used for content edits and we do not allow editing stream with content. I too was confused about this while changing the code.

Maybe we can move this in `if is_content_edited` block.

Reply:
Yeah, clarifying that, maybe in a follow-up commit, seems good.